crates/uv-auth/src/providers.rs

```diff
@@ -0,0 +1,42 @@
+use std::sync::LazyLock;
+
+use url::Url;
+
+use uv_static::EnvVars;
+
+use crate::Credentials;
+use crate::realm::Realm;
+
+/// The [`Realm`] for the Hugging Face platform.
+static HUGGING_FACE_REALM: LazyLock<Realm> = LazyLock::new(|| {
+    let url = Url::parse("https://huggingface.co").expect("Failed to parse Hugging Face URL");
+    Realm::from(&url)
+});
+
+/// The authentication token for the Hugging Face platform, if set.
+static HUGGING_FACE_TOKEN: LazyLock<Option<Vec<u8>>> = LazyLock::new(|| {
+    if std::env::var_os(EnvVars::UV_NO_HF_TOKEN).is_some() {
+        return None;
+    }
```

Comment:
We might want to do this after we've determined `HF_TOKEN` is populated (i.e., via `filter`) so we can include a debug or trace log indicating we ignored `HF_TOKEN` because of `UV_NO_HF_TOKEN`.